src/fidesops/task/graph_task.py

```diff
@@ -158,22 +200,43 @@ def pre_process_input_data(self, *data: List[Row]) -> NodeInput:
             )
 
         output: Dict[str, List[Any]] = {}
+
+        (
+            independent_field_mappings,
+            dependent_field_mappings,
+        ) = self.build_incoming_field_path_maps(group_dependent_fields)
+
+        if group_dependent_fields:
+            output[FIDESOPS_GROUPED_INPUTS] = []
         for i, rowset in enumerate(data):
             collection_address = self.input_keys[i]
-            field_mappings: List[
-                Tuple[FieldPath, FieldPath]
-            ] = self.incoming_field_path_map[collection_address]
 
             logger.info(
                 f"Consolidating incoming data into {self.traversal_node.node.address} from {collection_address}."
             )
             for row in rowset:
-                for foreign_field_path, local_field_path in field_mappings:
+                # Consolidate lists of independent field inputs
+                for foreign_field_path, local_field_path in independent_field_mappings[
+                    collection_address
+                ]:
                     new_values: List = consolidate_query_matches(
                         row=row, target_path=foreign_field_path
                     )
                     if new_values:
                         append(output, local_field_path.string_path, new_values)
+
+                # Separately group together dependent inputs if applicable
+                if dependent_field_mappings[collection_address]:
+                    grouped_data: Dict[str, Any] = {}
+                    for (
+                        foreign_field_path,
+                        local_field_path,
+                    ) in dependent_field_mappings[collection_address]:
+                        dependent_values: List = consolidate_query_matches(
+                            row=row, target_path=foreign_field_path
+                        )
+                        grouped_data[local_field_path.string_path] = dependent_values
+                    output[FIDESOPS_GROUPED_INPUTS].append(grouped_data)
```

Comment:
This is a change to the fidesops core, grouped fields can be separated out under a key that shouldn't overlap with a field on the user's collection/api endpoint.  This change should be added here rather than downstream because this is where we have the original rows of data from the upstream collections, and we can make sure two fields from a single incoming row are kept together.  Additionally, this opens it up to help build more extensive "AND" queries on our databases in the future, not just saas configs.

As a first step though, we're just setting up the ability to specify this on a saas config, but putting the foundation here makes it extensible.